freqtrade/exchange/binance.py

```diff
@@ -0,0 +1,549 @@
+            if h['fatal']:
+                raise OperationalException('{message} params=({currency})'.format(
+                    message=str(e),
+                    currency=currency))
+
+        return float(data['free'] or 0.0)
+
+    def get_balances(self) -> List[Dict]:
+
+        try:
+            data = _API.get_account()
+        except Exception as e:
+            h = Binance._handle_exception(e)
+            if h['fatal']:
+                raise OperationalException('{message}'.format(message=str(e)))
+
+        balances = data['balances']
+
+        currency_balances = []
+        for currency in balances:
+            balance = {}
+
+            if float(currency['free']) == 0 and float(currency['locked']) == 0:
+                continue
+            balance['Currency'] = currency.pop('asset')
+            balance['Available'] = currency.pop('free')
+            balance['Pending'] = currency.pop('locked')
+            balance['Balance'] = float(balance['Available']) + float(balance['Pending'])
+
+            currency_balances.append(balance)
+
+        return currency_balances
+
+    def get_ticker(self, pair: str, refresh: Optional[bool] = True) -> dict:
+
+        symbol = self._pair_to_symbol(pair)
+
+        api_try = True
+        tries = 0
+        max_tries = 1
+
+        while api_try and tries < max_tries:
+            try:
+                tries = tries + 1
+                data = _API.get_ticker(symbol=symbol)
+            except Exception as e:
+                h = Binance._handle_exception(e)
+                if h['fatal']:
+                    raise OperationalException('{message} params=({pair} {refresh})'.format(
+                        message=str(e),
+                        pair=pair,
+                        refresh=refresh))
+                api_try = h['retry']
+                max_tries = h['retry_max']
+                sleep(0.1)
+
+        return {
+            'bid': float(data['bidPrice']),
+            'ask': float(data['askPrice']),
+            'last': float(data['lastPrice']),
+        }
+
+    def get_ticker_history(self, pair: str, tick_interval: int) -> List[Dict]:
+
+        if tick_interval == 1:
+            INTERVAL_ENUM = KLINE_INTERVAL_1MINUTE
+        elif tick_interval == 5:
+            INTERVAL_ENUM = KLINE_INTERVAL_5MINUTE
+        elif tick_interval == 30:
+            INTERVAL_ENUM = KLINE_INTERVAL_30MINUTE
+        elif tick_interval == 60:
+            INTERVAL_ENUM = KLINE_INTERVAL_1HOUR
+        elif tick_interval == 1440:
+            INTERVAL_ENUM = KLINE_INTERVAL_1DAY
+        else:
+            raise ValueError('Cannot parse tick_interval: {}'.format(tick_interval))
+
+        symbol = self._pair_to_symbol(pair)
+
+        api_try = True
+        tries = 0
+        max_tries = 1
+
+        while api_try and tries < max_tries:
+            try:
+                tries = tries + 1
+                data = _API.get_klines(symbol=symbol, interval=INTERVAL_ENUM)
```

Comment:
Is there a reason you used get_klines instead of get_historical_klines with configurable start and end dates ?

Reply:
because the bot does not utilise configurable start or end dates and does not use that functionality. This is not used in Bittrex support either

Reply:
ok fair enough 👍

Reply:
BTW are you on slack ?

Reply:
For work yes, in my private time not so often, but I can be if you guys want to discuss some more about this pull request. I'm however leaving for some vacation now. I'll be back in about 1,5 week.